From d2ac12da7ed85ac3e20ff07cf65bedf529d2f4e6 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:41:26 -0400
Subject: [alsa] core: create write-only attribute macros for devices and drivers

Message-id: <140972823573-425-patchset-jkysela@redhat.com>
Patchwork-id: 91121
O-Subject: [RHEL 7.1 PATCH 424/442] ALSA: driver core: create write-only attribute macros for devices and drivers
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

This creates the macros DRIVER_ATTR_WO() and DEVICE_ATTR_WO() for
write-only attributes for drivers and devices.

Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

(picked from kernel: 1130c55c757660ac83bd2de220f08580c56b6855)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/device.h b/include/linux/device.h
index 9efbedb..2b10542 100644
--- a/include/linux/device.h
+++ b/include/linux/device.h
@@ -288,6 +288,8 @@ struct driver_attribute {
  struct driver_attribute driver_attr_##_name = __ATTR_RW(_name)
 #define DRIVER_ATTR_RO(_name) \
  struct driver_attribute driver_attr_##_name = __ATTR_RO(_name)
+#define DRIVER_ATTR_WO(_name) \
+ struct driver_attribute driver_attr_##_name = __ATTR_WO(_name)
 
 extern int __must_check driver_create_file(struct device_driver *driver,
      const struct driver_attribute *attr);
@@ -545,6 +547,8 @@ ssize_t device_store_bool(struct device *dev, struct device_attribute *attr,
  struct device_attribute dev_attr_##_name = __ATTR_RW(_name)
 #define DEVICE_ATTR_RO(_name) \
  struct device_attribute dev_attr_##_name = __ATTR_RO(_name)
+#define DEVICE_ATTR_WO(_name) \
+ struct device_attribute dev_attr_##_name = __ATTR_WO(_name)
 #define DEVICE_ULONG_ATTR(_name, _mode, _var) \
  struct dev_ext_attribute dev_attr_##_name = \
   { __ATTR(_name, _mode, device_show_ulong, device_store_ulong), &(_var) }
-- 
1.7.1